File: python/iris_example.py
import xgboost as xgb
import numpy as np
from sklearn import datasets
# detect if in interactive mode or running as a script
import __main__ as main
if hasattr(main, '__file__'):
	from xgbjson.xgbjson import XgbJSON
else:
	from python.xgbjson.xgbjson import XgbJSON

iris = datasets.load_iris()
tn = iris.target_names
fn = iris.feature_names
label = np.array([1 if tn[i] == 'versicolor' else 0 for i in iris.target])
fnames = [i.replace(' (cm)', '').replace(' ', '_') for i in fn]
dtrain = xgb.DMatrix(iris.data, label=label, feature_names=fnames)
param = {
	'max_depth': 2, 
	'eta': 0.1, 
	'objective': 'binary:logistic',
    'silent': 1
}
model = xgb.train(params=param, dtrain=dtrain, num_boost_round=100)
model.save_model('python/model.bin')
xgbjson = XgbJSON(model, na_value='null')
with open('python/model.js', 'w') as f:
    f.write(xgbjson.to_json())

File: R/iris_example.R
library(xgboost)
data(iris)
x <- as.matrix(iris[, 1:4])
y <- ifelse(iris$Species == 'versicolor', 1L, 0L)
dtrain <- xgb.DMatrix(x, label = y)
param <- list(
	max.depth = 2, 
	eta = 0.1, 
	objective = 'binary:logistic'
)
set.seed(0)
model <- xgb.train(params = param, data = dtrain, nrounds = 100)
source('R/xgbjson.R')
saveRDS(model, file = 'R/model.RDS')
feature_names = gsub('[.]', '_', tolower(colnames(x)))
xgbjson(
  model = model, 
  file = 'R/model.js', 
  fnames = feature_names, 
  na_value = 'null',
  regression = FALSE, 
  categoricals = NULL, 
  base_score = 0, 
  sparse_trained = FALSE
)
